packages/adblocker-webextension/src/index.ts

```diff
@@ -128,19 +139,98 @@ export function updateResponseHeadersWithCSP(
   return { responseHeaders };
 }
 
+const HTML_FILTERABLE_MIME_TYPES = new Set([
+  'application/javascript',
+  'application/json',
+  'application/mpegurl',
+  'application/vnd.api+json',
+  'application/vnd.apple.mpegurl',
+  'application/vnd.apple.mpegurl.audio',
+  'application/x-javascript',
+  'application/x-mpegurl',
+  'application/xhtml+xml',
+  'application/xml',
+  'audio/mpegurl',
+  'audio/x-mpegurl',
+]);
+export const MAXIMUM_RESPONSE_BUFFER_SIZE = 10 * 1024 * 1024;
+
+function getHeaderFromDetails(
+  details: OnHeadersReceivedDetailsType,
+  headerName: WebRequest.HttpHeadersItemType['name'],
+): string | undefined {
+  return details.responseHeaders?.find((header) => header.name.toLowerCase() === headerName)
+    ?.value;
+}
+
+export function shouldApplyReplaceSelectors(
+  request: Request<OnBeforeRequestDetailsType | OnHeadersReceivedDetailsType>,
+): boolean {
+  const details = request._originalRequestDetails as OnHeadersReceivedDetailsType;
+
+  // In case of undefined error of xhr/fetch and any kind of network activities
+  if (details.statusCode === 0) {
+    return false;
+  }
+
+  // If status code is non 2xx or 4xx, 5xx
+  const statusCodeScope = (details.statusCode / 100) >>> 0;
+  if (statusCodeScope !== 2 && statusCodeScope < 4) {
+    return false;
+  }
+
+  if (getHeaderFromDetails(details, 'content-disposition')?.startsWith('inline') === false) {
```

Comment:
Can you document what this does and when it can happen?

Reply:
Leaving ref [`c71579a` (#14)](https://github.com/seia-soto/adblocker/pull/14/commits/c71579a417157d5166b4efe58a8369aacd7baccb#diff-9545dc27a6748320d2b161f59633758bbfb8482494289e9181fcb5a85021b792R183)